src/psrsearch.jl

```diff
@@ -38,26 +36,25 @@ julia> z_n_binned([10., 0., 0., 0., 0.], 2)
 40.0
 ```
 """
-function z_n_binned(profile::Array{T,1}, n::Integer)::T where {T<:AbstractFloat}
+function z_n_binned(profile::AbstractVector{T}, n::Integer)::T where {T<:AbstractFloat}
```

Comment:
`Vector{T}` is simply an alias for `Array{T,1}`, in general I recommend to use that shorthand, it's easier to understand.  Similarly, `Matrix{T}` is an alias for `Array{T,2}`.

I suggest to use `AbstractVector` instead of `Vector` because in this way the function will be able to work with any subtype of `AbstractVector`, not just the standard `Vector`.  For example, GPU vectors are subtypes of `AbstractVector`, but they aren't `Vector`s

Reply:
Wow